apps/shipments/views/device.py

```diff
@@ -99,3 +104,36 @@ def telemetry(self, request, version, pk):
             telemetry_model_serializer.save()
 
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class SensorViewset(mixins.ListModelMixin, viewsets.GenericViewSet):
+    permission_classes = (IsDeviceOwnerOrShared,)
+
+    renderer_classes = (JSONRenderer,)
+
+    def get_serializer_class(self):
+        pass
+
+    def list(self, request, *args, **kwargs):
+        pk = self.kwargs['device_pk']
+        LOG.debug(f'Retrieving sensors for device with id: {pk}.')
+        log_metric('transmission.info', tags={'method': 'devices.sensors', 'module': __name__})
+
+        response = requests.get(
+            f'{settings.PROFILES_URL}/api/v1/device/{pk}/sensor',
+            params=request.query_params.dict()
+        )
+
+        try:
+            response_json = response.json()
+        except JSONDecodeError:
+            raise Custom500Error(detail="Invalid response returned from profiles.", status_code=503)
+
+        if response.status_code == 200:
+            links = {}
```

Comment:
These four lines could be combined using a dict comprehension (might still be multi lines but I prefer dict comprehensions to 1-line loops)